Create 2017-4-12-swarm-service-discovery.markdown

ajeetraina opened the pull request:
Added a short tutorial on Service Discovery under Swarm Mode

marcosnils commented on _posts/2017-01-28-swarm-stack-intro.markdown:
````diff
@@ -100,7 +100,7 @@ s/examplevotingapp_worker:latest
 
 ```.term1
-docker service ps voting_stack_vote
+docker service ps voting_stage_vote
````
Per https://github.com/play-with-docker/play-with-docker.github.io/pull/22/files#diff-649c2c995935407d8df16a950d7e9c98R60 the stack deployed is called `voting_stack`. I don't understand where this change comes from.

asiftasleem replied:
Oh I see. You are actually right

ajeetraina committed:
Create 2017-4-12-swarm-service-discovery.markdown

Added a short tutorial on Service Discovery

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
```diff
@@ -0,0 +1,185 @@
+
+
+{:.quiz}
+What features are available under Swarm Mode?
```
There's no marked answer.

ajeetraina replied:
Oops..I think I missed the answer section. Will add it.

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
```diff
@@ -0,0 +1,185 @@
+tags: [linux,operations]
+categories: beginner
+img: "swarm-service-discovery.png"
+terms: 2
```
You're adding two terms but only using one in the whole tutorial.

ajeetraina replied:
Good catch. Will fix it.

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
````diff
@@ -0,0 +1,185 @@
+
+From the first terminal, check the number of nodes in the swarm.
+
+```.term1
````
Maybe remove extra nodes to avoid confusion in the output?. As the one shown by the tutorial will only have two nodes.

ajeetraina replied:
Sure. I can go ahead and remove the extra nodes.

ajeetraina committed:
Update 2017-4-12-swarm-service-discovery.markdown

Added the changes as per the suggestion.

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
```diff
@@ -39,8 +39,6 @@ docker node ls
 sxn3hrguu3n41bdt9srhqbnva *  node1     Ready   Active        Leader
-tuhf7tmw4vxs3zq141x4ccgqg    node4     Ready   Active
-w8dwf7sbgv3sa1otwy4ptqa5p    node3     Ready   Active
 xsuyuqj2v254dy09t00w8we0t    node2     Ready   Active
```
We should remove node2 as well as the swarm has only one leader in this tutorial.

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
````diff
@@ -49,7 +47,7 @@ For example, the above command shows 4 nodes, the first one being the manager, a
 
 ```.term1
-docker network create -d overlay net1
+$docker network create -d overlay net1
````
incorrect character here?

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
````diff
@@ -96,7 +94,7 @@ Let's list the tasks of the wordpressdb service.
 ```
 
-You should get an output like the following one where the 2 tasks (replicas) of the service are listed.
+You should get an output like the following one where the 1 task  of the service are listed.
````
This sentence needs correction:

`You should get an output like the following one where 1 task of the service is listed`

ajeetraina committed:
Update 2017-4-12-swarm-service-discovery.markdown

ajeetraina committed:
Update 2017-4-12-swarm-service-discovery.markdown

I want to showcase Service Discovery across the cluster and hence other tasks might be running on the worker nodes. Hence, require atleast 2 nodes to demonstrate it.

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
```diff
@@ -0,0 +1,170 @@
+date:   2017-04-12
+author: "@ajeetsraina"
+tags: [linux,operations]
+categories: beginner
```
maybe intermediate?. Swarm concepts are for more advanced docker users usually.

ajeetraina replied:
Yeah, was thinking over it. Will keep it intermediate.

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
````diff
@@ -0,0 +1,170 @@
+
+```.term1
+docker node ls
+ID                           HOSTNAME  STATUS  AVAILABILITY  MANAGER STATUS
````
The command output should be inside ``` when using `.term1` or it'll be executed in the terminal.

ajeetraina replied:
Sure.

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
````diff
@@ -0,0 +1,170 @@
+
+```.term1
+docker network ls
+NETWORK ID          NAME                DRIVER              SCOPE
````
Output shouldn't be inside the ```.term1

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
````diff
@@ -0,0 +1,170 @@
+The output should be like the following one (your ID should be different though).
+
+```.term1
+ID            NAME                     MODE        REPLICAS  IMAGE
````
Output shouldn't be inside ```.term1.

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
````diff
@@ -0,0 +1,170 @@
+
+```.term1
+docker service ps wordpressdb
+ID                  NAME                IMAGE               NODE                DESIRED STATE
````
Output shouldn't be inside ```.term1

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
````diff
@@ -0,0 +1,170 @@
+
+```.term1
+docker service ls
+ID                  NAME                MODE                REPLICAS            IMAGE
````
Output shouldn't be inside ```.term1.

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
````diff
@@ -0,0 +1,170 @@
+
+```.term1
+docker ps
+CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS
````
Output shouldn't be inside ```.term1.

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
````diff
@@ -0,0 +1,170 @@
+try to reach out to wordpressab service using the service name.
+
+```.term1
+docker container exec -it 31b ping wordpressdb
````
Output shouldn't be inside ```.term1.

ajeetraina committed:
Update 2017-4-12-swarm-service-discovery.markdown

Remove .term1 and made it to intermediate. Do let me know if anything else is required to be changed.

ajeetraina committed:
Update 2017-4-12-swarm-service-discovery.markdown

I have added ```.term1 for commands and ``` for the output now.

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
```diff
@@ -0,0 +1,170 @@
+docker node ls
+ID                           HOSTNAME  STATUS  AVAILABILITY  MANAGER STATUS
+sxn3hrguu3n41bdt9srhqbnva *  node1     Ready   Active        Leader
+xsuyuqj2v254dy09t00w8we0t    node2     Ready   Active
```
`node2` needs to be removed. This tutorial doesn't use 2 nodes.

marcosnils reviewed (changes requested):
@ajeetraina seems like you're confused about how to use

\`\`\`.term1
\`\`\`
and  just plain

\`\`\`
\`\`\`

the first one should be used to put only commands that will be executed in the terminal when clicking them. The second one is just for outputs.

There are several places in the article where \`\`\`.term1 can be used and you're using \`\`\` instead.

ajeetraina committed:
Update 2017-4-12-swarm-service-discovery.markdown

Added 2 terminals and have used both under the tutorial. I have instructions to create 1 master and 1 worker node. hence, I need 2 node cluster to demonstrate Service Discovery.

ajeetraina committed:
Update 2017-4-12-swarm-service-discovery.markdown

marcosnils commented on _posts/2017-4-12-swarm-service-discovery.markdown:
````diff
@@ -0,0 +1,209 @@
+docker swarm init --advertise-addr $(hostname -i)
+```
+
+This node becomes a master node. The output displays a command to add a worker node to this swarm as shown below:
````
Is it relevant to the content of the article how to add a worker?. It's not mentioned anywhere else.

ajeetraina commented:
Hi,

As I am talking about Swarm Mode,  I wanted to demonstrate with at least 2
node cluster. As I want to demo that a container sitting on master node can
reach out to other container running the service by service name. That's
the whole purpose.

I added the instructions earlier,  will go ahead and add as an example.

Ajeet

+From the output above, copy the join command (*watch out for newlines*),
add the new instance and paste the command.This should join the new node to
the swarm mode and it becomes a worker node

On 13-Apr-2017 12:44 AM, "Marcos Nils" <notifications@github.com> wrote:

> *@marcosnils* commented on this pull request.
> ------------------------------
>
> In _posts/2017-4-12-swarm-service-discovery.markdown
> <https://github.com/franela/franela.github.io/pull/22#discussion_r111234484>
> :
>
> > +
> +## The application
> +
> +WordPress is an open-source content management system (CMS) based on PHP and MySQL.
> +It is a very simple containers application often used for demo purposes during meetup and conferences.
> +
> +
> +## Init your swarm
> +
> +Let's create a Docker Swarm first. Open up the first instance and initiate Swarm mode cluster.
> +
> +```.term1
> +docker swarm init --advertise-addr $(hostname -i)
> +```
> +
> +This node becomes a master node. The output displays a command to add a worker node to this swarm as shown below:
>
> Is it relevant to the content of the article how to add a worker?. It's
> not mentioned anywhere else.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/franela/franela.github.io/pull/22#pullrequestreview-32466015>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AATIiEdyu598U8jdrASp2u7DkLVY9_OEks5rvSKXgaJpZM4M67JA>
> .
>

marcosnils commented:
> As I am talking about Swarm Mode,  I wanted to demonstrate with at least 2
> node cluster. As I want to demo that a container sitting on master node can
> reach out to other container running the service by service name. That's
> the whole purpose.

Ok, makes sense. If that is the intention then maybe we can update the tutorial to use two terms and show how the services are moved across the different nodes. As it is currently, only one swarm node is being used.

ajeetraina committed:
Update 2017-4-12-swarm-service-discovery.markdown

I have revisited the whole setup, added term2  and shown output clearly on both the nodes to demonstrate Service Discovery.
